variables.tf

```diff
@@ -57,6 +57,24 @@ variable "agents_min_count" {
   description = "Minimum number of nodes in a pool"
 }
 
+variable "http_proxy_config" {
+  type = object({
+    http_proxy  = optional(string)
+    https_proxy = optional(string)
+    no_proxy    = optional(list(string))
+    trusted_ca  = optional(string)
+  })
+  default     = null
+  description = <<-EOT
+    optional(object({
+      http_proxy  = (Optional) The proxy address to be used when communicating over HTTP. Changing this forces a new resource to be created.
```

Comment:
"Changing this forces a new resource to be created." is not true anymore.

References:
https://github.com/Azure/AKS/issues/3524
https://github.com/MicrosoftDocs/azure-docs/commit/0bb10c7f3b608a541c48f2c1a9bf0574a22a8cd1

This will probably change soon upstream in the provider. Hold before merging.